base/commands/migration/start_stages.go

```diff
@@ -0,0 +1,219 @@
+			SuccessMsg:  "Started the migration",
+			FailureMsg:  "Could not start the migration",
+			Func:        st.startStage(ctx, ec),
+		},
+		{
+			ProgressMsg: "Migrating the cluster",
+			SuccessMsg:  "Migrated the cluster",
+			FailureMsg:  "Could not migrate the cluster",
+			Func:        st.migrateStage(ctx, ec),
+		},
+	}
+}
+
+func (st *StartStages) connectStage(ctx context.Context, ec plug.ExecContext) func(stage.Statuser) error {
+	return func(status stage.Statuser) error {
+		var err error
+		st.ci, err = ec.ClientInternal(ctx)
+		if err != nil {
+			return err
+		}
+		st.startQueue, err = st.ci.Client().GetQueue(ctx, StartQueueName)
+		if err != nil {
+			return err
+		}
+		st.statusMap, err = st.ci.Client().GetMap(ctx, MakeStatusMapName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateTopic, err = st.ci.Client().GetTopic(ctx, MakeUpdateTopicName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateMsgChan = make(chan UpdateMessage)
+		st.topicListenerID, err = st.updateTopic.AddMessageListener(ctx, st.topicListener)
+		return err
+	}
+}
+
+func (st *StartStages) topicListener(event *hazelcast.MessagePublished) {
+	var u UpdateMessage
+	err := json.Unmarshal(event.Value.(serialization.JSON), &u)
+	if err != nil {
+		panic(fmt.Errorf("receiving update from migration cluster: %w", err))
+	}
+	st.updateMsgChan <- u
+}
+
+func (st *StartStages) startStage(ctx context.Context, ec plug.ExecContext) func(stage.Statuser) error {
+	return func(stage.Statuser) error {
+		cb, err := makeConfigBundle(st.configDir, st.migrationID)
+		if err != nil {
+			return err
+		}
+		if err = st.startQueue.Put(ctx, cb); err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+		if isTerminal, err := st.handleUpdateMessage(ctx, ec, <-st.updateMsgChan); isTerminal {
+			return err
+		}
+		return nil
+	}
+}
+
+func makeConfigBundle(configDir, migrationID string) (serialization.JSON, error) {
+	var cb ConfigBundle
+	cb.MigrationID = migrationID
+	if err := cb.Walk(configDir); err != nil {
+		return nil, err
+	}
+	b, err := json.Marshal(cb)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (st *StartStages) migrateStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(stage.Statuser) error {
+		for {
+			select {
+			case msg := <-st.updateMsgChan:
+				if isTerminal, err := st.handleUpdateMessage(ctx, ec, msg); isTerminal {
+					return err
+				}
+			case <-ctx.Done():
+				if err := ctx.Err(); err != nil {
+					if errors.Is(err, context.DeadlineExceeded) {
+						return timeoutErr
+					}
+					return fmt.Errorf("migration failed: %w", err)
+				}
+			}
+		}
+	}
+}
+
+func (st *StartStages) handleUpdateMessage(ctx context.Context, ec plug.ExecContext, msg UpdateMessage) (bool, error) {
+	ec.PrintlnUnnecessary(msg.Message)
+	if slices.Contains([]Status{StatusComplete, StatusFailed, StatusCanceled}, msg.Status) {
```

Comment:
Update message is for updating and showing the real time updates. Not just for detecting if the migration is ended. So I would expect you to show migration status and completion percentage and update them in every update message.

Reply:
The following methods can be used to update the "In Progress" message during migration:
```
status.SetProgress()
status.SetRemainingDuration()
```

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/367/commits/dc45c7ab26cd55a6dc92f3ee424aefff7d1de0e8